consensus/core/src/commit_observer.rs

```diff
@@ -76,7 +80,14 @@ impl CommitObserver {
 
         for committed_sub_dag in committed_sub_dags.into_iter() {
             // Failures in sender.send() are assumed to be permanent
-            if let Err(err) = self.sender.send(committed_sub_dag.clone()) {
+            if let Err(err) = self.sender.send(ConsensusOutput::new(
+                committed_sub_dag.clone(),
+                self.leader_schedule
+                    .leader_swap_table
+                    .read()
+                    .reputation_scores
```

Comment:
You probably have thought / discussed this but I think associating the current reputation score with all commits here can be too confusing. For example the score could have be computed at index 600 for commits 301~600, and the commits here are for index 596~605. We should either:
1. Associate the score only with commit index 600. Leave the Vec empty otherwise.
2. Send `(Vec<CommittedSubDag>, Score)` to execution

Reply:
resolved offline to associate the current scores only with a subdag, otherwise it will be empty.